examples/common/hf_fsdp.py

```diff
@@ -104,6 +105,9 @@ def prepare_hf_causal_lm_model_for_fsdp(model: PreTrainedModel) -> None:
     HuggingFace for decoder-only LLMs.
     """
     causal_base_model = hf_get_causal_base_model(model)
+    # OPT has an extra layer of wrapping, so special case here
+    if isinstance(causal_base_model, OPTDecoder):
+        model.model._fsdp_wrap = False
```

Comment:
damn, this was supposed to be caught by
```python
    decoder_attrs = ('transformer', 'model.decoder', 'gpt_neox')
    return findattr(model, decoder_attrs)
```

I wonder why this isn't working

Reply:
opt has an extra layer, its ComposerHFCausalLM.OPTForCausalLM.OPTModel.OPTDecoder vs ComposerHFCausalLM.GPT2LMHeadModel.GPT2Model. So I'm explicitly setting false for the `OPTModel`, while the existing code is setting false for the `OPTDecoder`.

Reply:
ohh, got it